proposals/lip-0051.md

```diff
@@ -769,14 +770,222 @@ if the above send function does not fail:
+                    },
+                    "lockedBalances": {
+                        "type": "array",
+                        "fieldNumber": 4,
+                        "items": {
+                            "type": "object",
+                            "required": ["moduleID", "amount"],
+                            "properties": {
+                                "moduleID": {
+                                    "dataType": "uint32",
+                                    "fieldNumber": 1
+                                },
+                                "amount": {
+                                    "dataType": "uint64",
+                                    "fieldNumber": 2
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "supplySubstore": {
+            "type": "array",
+            "fieldNumber": 1,
+            "items": {
+                "type": "object",
+                "required": ["localID", "totalSupply"],
+                "properties": {
+                    "localID": {
+                        "dataType": "uint32",
+                        "fieldNumber": 1
+                    },
+                    "totalSupply": {
+                        "dataType": "uint64",
+                        "fieldNumber": 1
+                    }
+                }
+            }
+        },
+        "escrowSubstore": {
+            "type": "array",
+            "fieldNumber": 1,
+            "items": {
+                "type": "object",
+                "required": ["escrowChainID", "tokenLocalID", "amount"],
+                "properties": {
+                    "escrowChainID": {
+                        "dataType": "bytes",
+                        "fieldNumber": 1
+                    },
+                    "localID": {
+                        "dataType": "uint32",
+                        "fieldNumber": 2
+                    },
+                    "amount": {
+                        "dataType": "uint64",
+                        "fieldNumber": 3
+                    }
+                }
+            }
+        },
+        "availableLocalIDSubstore": {
+            "dataType": "uint32",
+            "fieldNumber": 1
+        },
+        "terminatedEscrowSubstore": {
+            "type": "array",
+            "fieldNumber": 1,
+            "items": {
+                "type": "uint32"
+            }
+        }
+    }
+}
+```
+
+
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the Token module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisTokenStoreSchema` schema, given above.
+
+* Initial checks:
```

Comment:
It is self-explanatory to me but to be consistent with following bullet points, we should add `genesisBlockAssetObject.` to every property that is referenced in these initial checks.

Reply:
reworded (tried to avoid the repetition though)